webapp/plugins/index.js

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+// EXPERIMENTAL - SUBJECT TO CHANGE
+
+import store from 'stores/redux_store.jsx';
+
+global.window.plugins = {};
+global.window.plugins.components = {};
+
+export function registerComponents(components) {
+    global.window.plugins.components = {...components, ...global.window.plugins.components};
```

Comment:
I would put them into the redux store. That way we can possibly use that cool tracking and other tools with it.